File: SourceAFIS/Extraction/Minutiae/StandardDpiScaling.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SourceAFIS.General;
using SourceAFIS.Meta;
using SourceAFIS.Templates;

namespace SourceAFIS.Extraction.Minutiae
{
    public sealed class StandardDpiScaling
    {
        [DpiAdjusted]
        [Parameter(Lower = 500, Upper = 500)]
        public int DpiScaling = 500;

        public DetailLogger.Hook Logger = DetailLogger.Null;

        public void Scale(TemplateBuilder template)
        {
            float dpiFactor = 500 / (float)DpiScaling;
            foreach (var minutia in template.Minutiae)
                minutia.Position = Calc.Round(Calc.Multiply(dpiFactor, minutia.Position));
            Logger.Log(template);
        }
    }
}

File: java/sourceafis/src/sourceafis/extraction/minutiae/StandardDpiScaling.java
/**
 * @author Veaceslav Dubenco
 * @since 19.10.2012
 */
package sourceafis.extraction.minutiae;

import sourceafis.general.Calc;
import sourceafis.general.DetailLogger;
import sourceafis.meta.DpiAdjusted;
import sourceafis.meta.Parameter;
import sourceafis.templates.Minutia;
import sourceafis.templates.TemplateBuilder;

/**
 * 
 */
public final class StandardDpiScaling {
	@DpiAdjusted
	@Parameter(lower = 500, upper = 500)
	public int DpiScaling = 500;

	public DetailLogger.Hook Logger = DetailLogger.off;

	public void Scale(TemplateBuilder template) {
		float dpiFactor = 500 / (float) DpiScaling;
		for (Minutia minutia : template.minutiae)
			minutia.Position = Calc.Multiply(dpiFactor, minutia.Position);
		Logger.log(template);
	}
}
